From bf5c074384680c5e35d18cf904af541dd9203010 Mon Sep 17 00:00:00 2001
From: Lennart Poettering <lennart@poettering.net>
Date: Fri, 24 Jul 2015 02:10:32 +0200
Subject: [PATCH] journalctl: properly detect empty journal files

When we encounter a journal file with exactly zero entries, print a nice
message and exit, and don't print a weird error message.

(cherry picked from commit 02ab86c732576a71179ce12e97d44c289833236d)
---
 src/journal/journalctl.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/journal/journalctl.c b/src/journal/journalctl.c
index c26cc00f51..388ae11fee 100644
--- a/src/journal/journalctl.c
+++ b/src/journal/journalctl.c
@@ -2057,6 +2057,10 @@ int main(int argc, char *argv[]) {
                 log_error_errno(r, "Failed to iterate through journal: %m");
                 return EXIT_FAILURE;
         }
+        if (r == 0) {
+                printf("-- No entries --\n");
+                goto finish;
+        }
 
         if (!arg_follow)
                 pager_open_if_enabled();
